display_vk/plugin.cpp

```diff
@@ -0,0 +1,228 @@
+#define GLFW_INCLUDE_VULKAN
+#include <GLFW/glfw3.h>
+
+#define VMA_IMPLEMENTATION
+#include "common/data_format.hpp"
+#include "common/phonebook.hpp"
+#include "common/switchboard.hpp"
+#include "common/threadloop.hpp"
+#include "common/vk_util/display_sink.hpp"
+#include "common/vk_util/vulkan_utils.hpp"
+
+using namespace ILLIXR;
+
+class display_vk : public display_sink {
+public:
+    display_vk(const phonebook* const pb)
+        : sb{pb->lookup_impl<switchboard>()} { }
+
+    /**
+     * @brief This function sets up the GLFW and Vulkan environments. See display_sink::setup().
+     */
+    void setup() {
+        setup_glfw();
+        setup_vk();
+    }
+
+    /**
+     * @brief This function recreates the Vulkan swapchain. See display_sink::recreate_swapchain().
+     */
+    virtual void recreate_swapchain() override {
+        vkb::SwapchainBuilder swapchain_builder{vkb_device};
+        auto                  swapchain_ret = swapchain_builder.set_old_swapchain(vk_swapchain)
+                                 .set_desired_present_mode(VK_PRESENT_MODE_FIFO_KHR)
+                                 .set_desired_extent(display_params::width_pixels, display_params::height_pixels)
+                                 .build();
+        if (!swapchain_ret) {
+            ILLIXR::abort("Failed to create Vulkan swapchain. Error: " + swapchain_ret.error().message());
+        }
+        vkb_swapchain          = swapchain_ret.value();
+        vk_swapchain           = vkb_swapchain.swapchain;
+        swapchain_images       = vkb_swapchain.get_images().value();
+        swapchain_image_views  = vkb_swapchain.get_image_views().value();
+        swapchain_image_format = vkb_swapchain.image_format;
+        swapchain_extent       = vkb_swapchain.extent;
+    }
+
+    /**
+     * @brief This function polls GLFW events. See display_sink::poll_window_events().
+     */
+    void poll_window_events() override {
+        should_poll = true;
+    }
+
+private:
+    /**
+     * @brief Sets up the GLFW environment.
+     *
+     * This function initializes the GLFW library, sets the window hints for the client API and resizability,
+     * and creates a GLFW window with the specified width and height.
+     *
+     * @throws runtime_error If GLFW initialization fails.
+     */
+    void setup_glfw() {
+        if (!glfwInit()) {
+            ILLIXR::abort("Failed to initalize glfw");
+        }
+
+        glfwWindowHint(GLFW_CLIENT_API, GLFW_NO_API);
+        glfwWindowHint(GLFW_RESIZABLE, GL_FALSE);
+
+        window =
+            glfwCreateWindow(display_params::width_pixels, display_params::height_pixels, "Vulkan window", nullptr, nullptr);
```

Comment:
Can we rename this to something like "ILLIXR Extended Vulkan Window" or something similar? And then incorporate preprocessor ifdefs to disable the `xlib_gl_extended_window` in `runtime_impl` when Vulkan is enabled. Right now just seems to be an empty window for me. Otherwise LGTM.

Reply:
I just renamed it to the eyebuffer window. I enabled debugview in again, so we would still need the extended window for that to work properly.

Reply:
I think debugview has its own GLFW/GL context and is able to run fine (at least, it doesn't seem to consume anything provided by the extended window, and it runs fine for me when I comment it out manually in runtime_impl. What does the behavior look like for you?

Reply:
Yes, you're right. Fixed.